source/code/plugins/in_mysql_workload.rb

```diff
@@ -0,0 +1,47 @@
+module Fluent
+  class MysqlWorkloadInput < Fluent::Input
+    Plugin.register_input('mysql_workload', self)
+
+    # Define `router` method to support v0.10.57 or earlier
+    unless method_defined?(:router)
+      define_method("router") { Fluent::Engine }
+    end
+
+    def initialize
+      require_relative 'mysql_workload_lib'
+      super
+    end
+
+    config_param :host, :string, :default => 'localhost'
+    config_param :port, :integer, :default => 3306
+    config_param :username, :string, :default => 'root'
+    config_param :password, :string, :default => nil, :secret => true
+    config_param :database, :string, :default => nil
+    config_param :encoding, :string, :default => 'utf8'
+    config_param :interval, :time, :default => '1m'
+    config_param :tag, :string
+
+    def configure(conf)
+      super
+    end
+
+    def start
+      @thread = Thread.new(&method(:run))
+    end
+
+    def shutdown
+      Thread.kill(@thread)
```

Comment:
connection will eventually time out and be destroyed but I agree that it would be better to add explicit release of the resource. Fixing now.